exporter/datasetexporter/config.go

```diff
@@ -47,8 +62,12 @@ func (c *Config) Unmarshal(conf *confmap.Conf) error {
 		c.APIKey = os.Getenv("DATASET_API_KEY")
```

Comment:
you can provide the api key without going to the env vars, and use a config provider to read from env vars. Secrets typically use the configopaque package.

Reply:
I have changed it to `configopaque.String`.

About the env vars. It's one of the requirements since some people are still using them. It's also mentioned as recommended solution - https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#portable-code.

Reply:
Fixed.